File: spring-data-requery/src/test/java/org/springframework/data/requery/domain/superclass/Related.java
package org.springframework.data.requery.domain.superclass;

import io.requery.Entity;
import io.requery.Key;

/**
 * org.springframework.data.requery.domain.superclass.Related
 *
 * @author debop
 * @since 18. 6. 4
 */
@Entity
public interface Related {

    @Key
    Long getId();

}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/domain/superclass/model/Related.kt
package org.springframework.data.requery.kotlin.domain.superclass.model

import io.requery.Entity
import io.requery.Key
import io.requery.Persistable
import java.io.Serializable

/**
 * org.springframework.data.requery.kotlin.domain.superclass.model.Related
 *
 * @author debop
 */
@Entity
interface Related: Persistable, Serializable {

    @get:Key
    val id: Long

    var attribute: String
}